apps/api/src/secret/service/secret.service.ts

```diff
@@ -490,6 +490,43 @@ export class SecretService {
 
     return response
   }
+  async getRevisionsOfSecret(
+    user: User,
+    secretId: Secret['id'],
+    environmentId: Environment['id'],
+    page: number,
+    limit: number,
+    sort: string
+  ) {
+    //check access to secret
+    await this.authorityCheckerService.checkAuthorityOverSecret({
+      userId: user.id,
+      entity: { id: secretId },
+      authority: Authority.READ_SECRET,
+      prisma: this.prisma
+    })
+
+    await this.authorityCheckerService.checkAuthorityOverEnvironment({
+      userId: user.id,
+      entity: { id: environmentId },
+      authority: Authority.READ_ENVIRONMENT,
+      prisma: this.prisma
+    })
+
+    // get the revisions
+    const revisions = await this.prisma.secretVersion.findMany({
+      where: {
+        secretId: secretId,
+        environmentId: environmentId
+      },
+      skip: page * limit,
+      take: limit,
+      orderBy: {
+        [sort]: 'desc'
```

Comment:
Hey @anudeeps352 I think we need to fix this first!

Reply:
Ahh I'll check it out asap

Reply:
I have made the mod rn

Reply:
We don't want to take the input of the field we want to sort by since it will always be `version`: 
- https://github.com/keyshade-xyz/keyshade/pull/303/files#diff-504a3762c691dc3814c053b3815ff9b1f1a8eb54cde844d882d9ea06779eca84R113
- https://github.com/keyshade-xyz/keyshade/pull/303/files#diff-dee38177617754972e3d3f727d7f1536566d7a784b7ffdda74aa97d6eec4cbc5R525-R527

We just want to take in the sorting order as the input from the user. Value of `asc` or `desc`, `desc` being the default.

Reply:
Also, the tests are failing

Reply:
I think it would be better for me to pull the latest changes and do the mods over again